python/ql/test/library-tests/PointsTo/regressions/wrong/classmethod/test.py

```diff
@@ -0,0 +1,29 @@
+# See https://github.com/Semmle/ql/issues/3113
+def some_decorator(func):
+    print("this could be tricky for our analysis")
+    return func
+
+class Foo(object):
+
+    @classmethod
+    def no_problem(cls):
+        check(cls) # analysis says 'cls' can only point-to Class Foo
+
+    @some_decorator
+    @classmethod
+    def problem_through_instance(cls):
+        # Problem is that our analysis says that 'cls' can point to EITHER the
+        # Class Foo (correct) or an instance of Foo (wrong)
+        check(cls)
+
+    @some_decorator
+    @classmethod
+    def problem_through_class(cls):
+        check(cls) # same as above
+
```

Comment:
Does the issue arise if the decorators are swapped?

Reply:
yes. Just added a test for this though :+1:

Reply:
Nice!